kuka_rsi_hw_interface/src/rsi_command_handler.cpp

```diff
@@ -0,0 +1,325 @@
+  bool isBaseLessNode = false;
+  bool isNoMoreParam = false;
+  xml::XMLString node_param;
+  while (!isNoMoreParam && (size_t)(buffer_it - buffer) < buffer_size) {
+    // fast forward to the next non-space character
+    for (; *buffer_it == ' '; buffer_it++) {
+    }
+    if ((size_t)(buffer_it - buffer) >= buffer_size) {
+      throw std::range_error("Out of the buffers range");
+    }
+    // Check for the nodes hadders end characters
+    if (*buffer_it == '/' && *(buffer_it + 1) == '>') {
+      isBaseLessNode = true;
+      isNoMoreParam = true;
+    } else if (*buffer_it == '>') {
+      isNoMoreParam = true;
+    } else {
+      if (!element.IsParamNameValid(node_param, buffer_it)) {
+        auto i = std::sprintf(err_buf_, "The detected parameter (");
+        std::snprintf(err_buf_ + i, node_param.length_ + 1, "%s", node_param.data_ptr_);
+        i += node_param.length_;
+        std::sprintf(
+          err_buf_ + i, ") does not match with any of the %s elements parameters.",
+          element.GetName().c_str());
+        throw std::logic_error(err_buf_);
+      }
+      if (*buffer_it != '=') {
+        auto i = sprintf(err_buf_, "In \"");
+        std::snprintf(err_buf_ + i, node_param.length_ + 1, "%s", node_param.data_ptr_);
+        i += node_param.length_;
+        std::sprintf(err_buf_ + i, "\" param syntax error found");
+        throw std::logic_error(err_buf_);
+      }
+      buffer_it++;
+      if (*buffer_it != '"') {
+        auto i = sprintf(err_buf_, "In \"");
+        std::snprintf(err_buf_ + i, node_param.length_ + 1, "%s", node_param.data_ptr_);
+        i += node_param.length_;
+        std::sprintf(err_buf_ + i, "\" param syntax error found");
+        throw std::logic_error(err_buf_);
+      }
+      buffer_it++;
+      if (!element.CastParam(node_param, buffer_it)) {
+        auto i = std::sprintf(err_buf_, "Could not cast the ");
+        std::snprintf(err_buf_ + i, node_param.length_ + 1, "%s", node_param.data_ptr_);
+        i += node_param.length_;
+        std::sprintf(
+          err_buf_ + i, " param into the %s elements parameter list.",
+          element.GetName().c_str());
+        throw std::logic_error(err_buf_);
+      }
+      if (*buffer_it != '"') {
+        auto i = sprintf(err_buf_, "In \"");
+        std::snprintf(err_buf_ + i, node_param.length_ + 1, "%s", node_param.data_ptr_);
+        i += node_param.length_;
+        std::sprintf(err_buf_ + i, "\" param syntax error found");
+        throw std::logic_error(err_buf_);
+      }
+      buffer_it++;
+      numOfParam++;
+    }
+  }
+
+  // Fast forward to close bracket
+  for (; (size_t)(buffer_it - buffer) < buffer_size && *buffer_it != '>'; buffer_it++) {
+  }
+  if ((size_t)(buffer_it - buffer) >= buffer_size) {
+    throw std::range_error("Out of the buffers range");
+  }
+  buffer_it++;
+
+  // Could not find all parameter, or found too much
+  if (isBaseLessNode && numOfParam != element.GetParams().size()) {
```

Comment:
why check isBaseLess here?

Reply:
I think I get it, that "value-type" parameters like <abc>1</abc> are parsed later, but you could add a comment here too explaining

Reply:
commented